src/data/moviesSlice.js

```diff
@@ -0,0 +1,27 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit"
+
+export const fetchMovies = createAsyncThunk('fetch-movies', async (apiUrl) => {
+    const response = await fetch(apiUrl)
+    return response.json()
+})
+
+const moviesSlice = createSlice({
```

Comment:
Shoud add a pagination through new state properties (hasMore, page), which allows for accumulating results from multiple pages. (This makes the code more robust and scalable).